### PR TITLE
Dist artifacts

### PR DESCRIPTION
**Pull request by Hoverbear**
###  What have you changed?

Based off #5776.

This creates a `make dist_artifacts` command for future use in our CI system to build release binaries and Docker images.

It also adds some shorthand commands for the tagging of Docker images, also for later CI use.

###  What is the type of the changes?

This is a feature for build.

###  How is the PR tested?

Manual test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

Yes.

> `make dist_artifacts` now creates compressed archives of TiKV binaries and Docker images.

###  Does this PR affect `tidb-ansible`?

Not yet but a future change will.

###  Refer to a related PR or issue link (optional)

#5776 

###  Any examples? (optional)

```bash
$ make dist_artifacts
$ make docker-tag
docker tag pingcap/tikv pingcap/tikv:dcbc8a45d9bd3fddb660bbb244f096343c849353
docker tag pingcap/tikv pingcap/tikv:v3.0.0-beta.1-638-gdcbc8a45
$ tree dist -h
dist
├── [183M]  tikv-docker.tar.gz
└── [176M]  tikv.tar.gz

0 directories, 2 files
```

### CODE REVIEW

File: scripts/gen-dockerfile.sh [hunk 1]
```diff
@@ -1,66 +1,94 @@
...
-fi
+# Install the system dependencies
+# Attempt to clean and rebuild the cache to avoid 404s
+cat <<EOT
+RUN yum clean all \
+	&& yum makecache \
+	&& yum update -y \
+	&& yum install -y tar wget git which file unzip python-pip openssl-devel \
+		make cmake3 gcc gcc-c++ libstdc++-static pkg-config psmisc gdb \
+		libdwarf-devel elfutils-libelf-devel elfutils-devel binutils-devel golang \
```
Comment (siddontang) [inline scripts/gen-dockerfile.sh]: em, do we need golang now?

Comment (Hoverbear) [reply scripts/gen-dockerfile.sh]: We might not! Let me try. :)

**Comment by zhouqiang-cl** [comment]
@overvenus PTAL

File: scripts/gen-dockerfile.sh [hunk 2]
```diff
@@ -99,5 +131,3 @@ EXPOSE 20160 20180
 
 ENTRYPOINT ["/tikv-server"]
 EOT
-
-cat .gitignore > .dockerignore
```
Comment (Hoverbear) [inline scripts/gen-dockerfile.sh]: @overvenus PTAL, I think it's OK to let these diverge so we can optimize our `.dockerignore` later.

**Comment by Hoverbear** [comment]
From the benchmarking

```
[bench_sysbench_per_pr-731] SUCCESS
Elapsed Time: 72.44 Mins

Version:
   tidb: 380216ed202e7fd57a7a8539ced58e51efc0e0ac(master) 2019-11-19 10:24:40
   tikv: 5c400b7de10d7c53eacace64c8c7c81c11918b55(master) 2019-11-19 10:48:51
   pd: 08e0143e83a2c42955bdfcbed4862f0339b9befc(master) 2019-11-18 01:22:10
       
Compare:
   tidb: https://github.com/pingcap/tidb/compare/77faf64...380216e
   tikv: https://github.com/tikv/tikv/compare/5c400b7...5c400b7
   pd: https://github.com/pingcap/pd/compare/08e0143...08e0143
       
test-1: < oltp_update_non_index >
   * QPS : 33063.40 ± 0.1856% (std=45.44) delta: 0.53% (p=0.166)
   * TPS : 33063.40 ± 0.1856% (std=45.44) delta: 0.53%
   * AvgMs : 7.74 ± 0.1292% (std=0.01) delta: -0.55%
   * PercentileMs99 : 20.74 ± 0.0000% (std=0.00) delta: 0.00%
           
test-2: < oltp_insert >
   * QPS : 23350.51 ± 0.9477% (std=182.11) delta: -0.58% (p=0.333)
   * TPS : 23350.51 ± 0.9477% (std=182.11) delta: -0.58%
   * AvgMs : 10.96 ± 0.9493% (std=0.09) delta: 0.55%
   * PercentileMs99 : 22.44 ± 1.0961% (std=0.20) delta: 0.74%
           
test-3: < oltp_read_write >
   * QPS : 41653.52 ± 0.2184% (std=61.74) delta: 0.09% (p=0.565)
   * TPS : 2073.25 ± 0.2117% (std=3.09) delta: -0.00%
   * AvgMs : 123.32 ± 0.1176% (std=0.09) delta: -0.07%
   * PercentileMs99 : 179.31 ± 2.1750% (std=2.42) delta: 0.37%
           
test-4: < oltp_write_only >
   * QPS : 41644.46 ± 0.9640% (std=226.79) delta: -0.08% (p=0.626)
   * TPS : 6920.82 ± 0.9646% (std=37.67) delta: -0.08%
   * AvgMs : 36.85 ± 0.6134% (std=0.17) delta: -0.06%
   * PercentileMs99 : 69.29 ± 0.0000% (std=0.00) delta: -1.79%
           
test-5: < oltp_update_index >
   * QPS : 20310.56 ± 0.8509% (std=118.82) delta: 0.48% (p=0.425)
   * TPS : 20310.56 ± 0.8509% (std=118.82) delta: 0.48%
   * AvgMs : 12.53 ± 1.0059% (std=0.08) delta: 0.02%
   * PercentileMs99 : 28.67 ± 0.0000% (std=0.00) delta: 0.00%
           
test-6: &lt…
```

**Pull request merged**